main.py

```diff
@@ -1,32 +1,31 @@
+from math import exp
```

Comment:
The changes in this file are mainly moving the imports to the top of the file and adding a feedback message to the user when the df is saved